dvc/repo/fetch.py

```diff
@@ -22,7 +43,36 @@ def fetch(
             remote=remote,
             jobs=jobs,
             recursive=recursive,
-        )["local"]
-        return self.cloud.pull(
-            used, jobs, remote=remote, show_checksums=show_checksums
         )
+
+        downloaded = 0
+        failed = 0
+
+        try:
+            downloaded += self.cloud.pull(
+                used["local"],
+                jobs,
+                remote=remote,
+                show_checksums=show_checksums,
+            )
+        except NoRemoteError:
+            if not used["repo"] and used["local"]:
+                raise
+
+        except DownloadError as exc:
+            failed += exc.amount
+
+        for dep in used["repo"]:
+            try:
+                out = dep.fetch()
+                downloaded += out.get_files_number()
+            except DownloadError as exc:
+                failed += exc.amount
+            except (CloneError, OutputNotFoundError) as exc:
+                failed += 1
+                logger.exception(str(exc))
```

Comment:
Why do that though? :slightly_smiling_face: How about
```suggestion
                logger.exception("failed to fetch data for '{}'".format(dep.stage.outs[0]))
```
exception will automatically print the last exception, there is no need to pass it again. Also, when you are passing it explicitly, you will get duplicate messages, which is wrong.

Reply:
:+1: